src/main/java/org/apache/maven/plugins/scripting/EvalMojo.java

```diff
@@ -73,8 +84,10 @@ public void execute()
          AbstractScriptEvaluator execute = constructExecute();
 
          Bindings bindings = new SimpleBindings();
+         bindings.put( "session", session );
          bindings.put( "project", project );
          bindings.put( "log", getLog() );
+         bindings.put( "servers", new Servers( session, settingsDecrypter ) );

```

Comment:
I have doubts about this one, `servers` has never been used as a variable. Better to stick to `settings` with resolved values.

Reply:
Do we have it already? I would like to avoid to decipher the whole settings each time (this is done only when needed) because it can slow down the execution a loooot for nothing in practise.
Happy to wrap it in a MavenScripting wrapper which would expose this kind of API (so script would use mavenScripting.servers.find(...))

Reply:
If I understand what you need and staying close to what Maven offers, I think you should focus on `session.getRepositorySession().getAuthenticationSelector().getAuthentication(RemoteRepository)`

Reply:
What I want is a Server#password in clear but I don't want to decipher all servers upfront. From what I saw, getSettings() is NOT clear there so I would have to "clearify" it before binding it which I want to avoid so I thought using a lazy simple accessor is more script friendly than exposing the session and letting user use plexus container to do lookups to decipher values.